lib/checkfunctions.cpp

```diff
@@ -683,3 +683,133 @@ void CheckFunctions::copyElisionError(const Token *tok)
                 "Using std::move for returning object by-value from function will affect copy elision optimization."
                 " More: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-return-move-local");
 }
+
+void CheckFunctions::useStandardLibrary()
+{
+    if (!mSettings->severity.isEnabled(Severity::style))
+        return;
+
+    for (const Scope& scope: mTokenizer->getSymbolDatabase()->scopeList) {
+        if (scope.type != Scope::ScopeType::eFor)
+            continue;
+
+        const Token *forToken = scope.classDef;
+        // for ( initToken ; condToken ; stepToken )
+        const Token* initToken = getInitTok(forToken);
+        if (!initToken)
+            continue;
+        const Token* condToken = getCondTok(forToken);
+        if (!condToken)
+            continue;
+        const Token* stepToken = getStepTok(forToken);
+        if (!stepToken)
+            continue;
+
+        // 1. we expect that idx variable will be initialized with 0
+        const Token* idxToken = initToken->astOperand1();
+        const Token* initVal = initToken->astOperand2();
+        if (!idxToken || !initVal || !initVal->hasKnownIntValue() || initVal->getKnownIntValue() != 0)
+            continue;
+        const auto idxVarId = idxToken->varId();
+        if (0 == idxVarId)
+            continue;
+
+        // 2. we expect that idx will be less of some variable
+        if (!condToken->isComparisonOp())
+            continue;
+
+        const auto checkBoundaryType = [](const Token* t, const bool rightOp) -> bool {
+            const auto *const till = rightOp ? ";" : ">"; // valid end of expression
+
+            bool result = nullptr != t;
+            for (; result && nullptr != t && t->str() == till; t = t->next()) {
+                if (t->isNumber()) {
```

Comment:
you have checked that `t->str() == till` therefore this condition `t->isNumber()` can't be true.

Reply:
Oh my... Shame on me! It should be until not equal `till`. Thank you